src/integrationTest/java/com/mongodb/hibernate/query/select/SortingSelectQueryIntegrationTests.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.query.select;
+
+import static com.mongodb.hibernate.internal.MongoConstants.MONGO_DBMS_NAME;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.hibernate.cfg.QuerySettings.DEFAULT_NULL_ORDERING;
+
+import com.mongodb.hibernate.internal.FeatureNotSupportedException;
+import java.util.Arrays;
+import java.util.List;
+import org.hibernate.query.NullPrecedence;
+import org.hibernate.query.SortDirection;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.ServiceRegistry;
+import org.hibernate.testing.orm.junit.Setting;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+@DomainModel(annotatedClasses = Book.class)
+class SortingSelectQueryIntegrationTests extends AbstractSelectionQueryIntegrationTests {
+
+    private static final List<Book> testingBooks = List.of(
+            new Book(1, "War and Peace", 1869, true),
+            new Book(2, "Crime and Punishment", 1866, false),
+            new Book(3, "Anna Karenina", 1877, false),
+            new Book(4, "The Brothers Karamazov", 1880, false),
+            new Book(5, "War and Peace", 2025, false));
+
+    private static List<Book> getBooksByIds(int... ids) {
+        return Arrays.stream(ids)
+                .mapToObj(id -> testingBooks.stream()
+                        .filter(c -> c.id == id)
+                        .findFirst()
+                        .orElseThrow(() -> new IllegalArgumentException("id does not exist: " + id)))
+                .toList();
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        getSessionFactoryScope().inTransaction(session -> testingBooks.forEach(session::persist));
+        getTestCommandListener().clear();
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"ASC", "DESC"})
```

Comment:
Maybe use `AstSortOrder`?

Reply:
That was my original usage, but @stIncMale told me that is internal stuff so we'd better avoid using it. see https://github.com/mongodb/mongo-hibernate/pull/90#discussion_r2076033023 for the previous solved comment.